enhancements/network/static-ip-addresses-vsphere.md

```diff
@@ -0,0 +1,560 @@
+---
+title: static-ip-addresses-vsphere
+authors:
+  - rvanderp3
+  - vr4manta
+reviewers: 
+  - JoelSpeed - machine API
+  - elmiko - machine API
+  - patrickdillon - installer
+  - jcpowermac - vSphere
+  - zaneb - networking
+  - 2uasimojo - hive
+approvers:
+  - JoelSpeed
+  - patrickdillon
+  - 2uasimojo
+api-approvers: 
+  - JoelSpeed
+creation-date: 2022-10-21
+last-updated: 2024-01-02
+tracking-link: 
+- https://issues.redhat.com/browse/OCPPLAN-9654
+see-also:
+  - /enhancements/installer/vsphere-ipi-zonal.md
+replaces:
+superseded-by:
+---
+
+# Static IP Addresses for vSphere IPI
+
+## Summary
+
+Static IP addresses are emerging as a common requirement in environments where
+the usage of DHCP violates corporate security guidelines.  Additionally, many 
+users which require static IPs also require the use of the IPI installer. 
+The proposal described in this enhacement discusses the implementation of
+of assiging static IPs at both day 0 and day 2.
+
+## Motivation
+
+Users of OpenShift would like the ability to provision vSphere IPI clusters with static IPs.
+
+- https://issues.redhat.com/browse/OCPPLAN-9654
+
+### User Stories
+
+As an OpenShift administrator, I want to provision nodes with static IP addresses so that I can comply with my organization's security requirements.
+
+As an OpenShift administrator, I want to provision static IP addresses with the IPI installer so that I can reduce the complexity of certifying tools required to provision OpenShift.
+
+As an OpenShift administrator, I want to scale nodes with static IPs so that I can meet capacity demands as well as respond to disaster recovery scenarios.
+
+### Goals
+
+- All nodes created during the installation are configured with static IPs.  
+  Rationale: Many environments, due to security policies, do not allow DHCP.
+
+- The IPI installation method is able to provide static IPs to the nodes
+  Rationale: Some users must qualify each tool used in their environment. 
+  Leveraging IPI greatly reduces the number of tools required to provision
+  a cluster.
+
+### Non-Goals
+
+- OpenShift will not be responsible for managing IP addresses.
+
+## Proposal
+
+### Static IPs Configured at Installation
+
+To faciliate the configuration of static IP address, network device configuration definitions are created for each node in the install-config.yaml. A `hosts` slice will be introduced to the installer platform specification to allow network device configurations to be specified for a nodes. 
+If a `hosts` slice is defined, enough `hosts` must be defined to cover the bootstrap, control plane, and compute nodes to be provisioned during the installation process.
+
+The bootstrap and control plane nodes will have their static IP addresses provisioned by CAPV.  Compute nodes will rely on the machine API to provision their IP addresses.
+
+Example of platform specification with IP addresses defined for bootstrap, control plane, and compute nodes:
+~~~yaml
```

Comment:
Is this API up to date with the latest changes from o/api?

Reply:
These example here is now up-to-date.